core/src/processing/feeder.cc

```diff
@@ -60,7 +58,9 @@ feeder::feeder(
 /**
  *  Destructor.
  */
-feeder::~feeder() {}
+feeder::~feeder() {
+  this->bthread::~bthread();
```

Comment:
Does this call really necessary? You set bthread destructor as virtual, so the work should be done...